systems/subscriber/test-agent/pkg/server/test_agent.go

```diff
@@ -0,0 +1,142 @@
+package server
+
+import (
+	"context"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	pb "github.com/ukama/ukama/systems/subscriber/test-agent/pb/gen"
+	"github.com/ukama/ukama/systems/subscriber/test-agent/pkg/storage"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const iccidPrefix = "890000"
+
+type TestAgentServer struct {
+	storage storage.Storage
+	pb.UnimplementedTestAgentServiceServer
+}
+
+func NewTestAgentServer(storage storage.Storage) *TestAgentServer {
+	return &TestAgentServer{
+		storage: storage,
+	}
+}
+
+func (s *TestAgentServer) GetSimInfo(ctx context.Context, req *pb.GetSimInfoRequest) (*pb.GetSimInfoResponse, error) {
```

Comment:
Does this mean before Acitivating Sim we need to call GetSimInfo to create a sim ?

Reply:
No. If the sim does already exist, then we activate it by  making an `ActivateSim` RPC call if we are directly talking to the agent  or a `ToggleSimStatus` RPC if we talking to the agent via Sim Manager. The `GetSim`  (previously `GetSimInfo` ) will only create a new sim info for the first time if the sim does not exist.